### ISSUE TITLE
Unprocessable Entity error on start

### ISSUE THREAD

**Issue by hardikdangar** [carlosflorencio/laracasts-downloader#139]
Hello,

I am having issue that i can't seem to understand. few days back it was working but now it stopped with this message.

ERROR: Client error response [url] https://laracasts.com/sessions [status code] 422 [reason phrase] Unprocessable Entity 

I have setup everything in .env and it is loading in start.php correctly. but above is printed in console when i run the command i tried php 7.3 and 8.0 in both cases it has same set of issues.

**Comment by AmirRezaM75** [comment]
@hardikdangar Probably same as https://github.com/carlosflorencio/laracasts-downloader/issues/136 and you didn't pull the latest changes

**Comment by hardikdangar** [comment]
should i try dev branch ? thats what you are suggesting ?

**Comment by AmirRezaM75** [comment]
No, just use lastest version of master.
``git pull origin master``

**Comment by hardikdangar** [comment]
I did that and removed vendor and composer install with php8.0 no luck error is still the same. is there a way i can debug this? i am assuming this has something to do with curl but not able to find where i can debug this.

**Comment by AmirRezaM75** [comment]
I've just tried to run script with invalid username and password and got this error
```
ERROR: Client error response [url] https://laracasts.com/sessions [status code] 422 [reason phrase] Unprocessable Entity
```
Make sure they are correct, also there is also a possibility that they are not sent via guzzle
All the logic is within here:

https://github.com/carlosflorencio/laracasts-downloader/blob/246e12ac98dbf15ef4c7ba3b1664a8b850e89e52/App/Http/Resolver.php#L76-L99

**Comment by hardikdangar** [comment]
Hey thanks for that,

it was something to do with php version but was able to fix that by switching php version to 8.0.

Now it works but gives following warning. so authentication is done and it starts downloading but i get this constant errors. and files downloaded are of zero bytes.

`PHP Warning:  curl_exec(): Cannot call the CURLOPT_PROGRESSFUNCTION in ~/laracasts-downloader/vendor/guzzlehttp/ringphp/src/Client/CurlHandler.php on line 89`

**Comment by AmirRezaM75** [comment]
Unfortunately it doesn't work with php 8 (https://github.com/carlosflorencio/laracasts-downloader/issues/132)
Try PHP 7.4
